argschema/argschema_parser.py

```
@@ -136,43 +152,77 @@ class ArgSchemaParser(object):
+                 input_source = None,
+                 output_sink = None,
                  logger_name=__name__):
         
         if schema_type is None:
             schema_type = self.default_schema
         if output_schema_type is None:
             output_schema_type = self.default_output_schema
-
+        
         self.schema = schema_type()
         self.logger = self.initialize_logger(logger_name,'WARNING')
         self.logger.debug('input_data is {}'.format(input_data))
 
         # convert schema to argparse object
-        p = utils.schema_argparser(self.schema)
+
+        #consolidate a list of the input and output source
+        #command line configuration schemas
+        io_schemas = []
+        for in_cfg in self.input_config_map:
+            io_schemas.append(in_cfg.ConfigSchema())
+        for out_cfg in self.output_config_map:
+            io_schemas.append(out_cfg.ConfigSchema())
+
+        #build a command line parser from the input schemas and configurations
+        p = utils.schema_argparser(self.schema,io_schemas)
         argsobj = p.parse_args(args)
         argsdict = utils.args_to_dict(argsobj)
         self.logger.debug('argsdict is {}'.format(argsdict))
 
-        if argsobj.input_json is not None:
-            result = self.schema.load(argsdict)
-            if 'input_json' in result.errors:
-                raise mm.ValidationError(result.errors['input_json'])
-            with open(result.data['input_json'], 'r') as j:
-                jsonargs = json.load(j)
-        else:
-            jsonargs = input_data if input_data else {}
+        #if you received an input_source, get the dictionary from there
+        if input_source is not None:
+            input_data = input_source.get_dict()
+        else: #see if the input_data itself contains an InputSource configuration use that
+            for InputSource in self.input_config_map:
+                try: 
+                    input_data = get_input(InputSource,input_data)
```

I think this should at least  issue a warning if input_data returns for multiple sources to let users know if they accidentally generated conflicting input data. Right now it will just be generated from the last InputSource that was valid.

---

This is a good point.. actually I think it should raise an exception in this case because it's inputs are not well defined.